### ISSUE TITLE
Hopefully, the next version will support shortcuts in cmd.exe panel

### ISSUE THREAD

**Issue by JiangBoy** [kingToolbox/WindTerm#585]
exp: 
ctrl+a  => cursor jump to line begin
ctrl+e  => cursor jump to line end
ctrl+w/ctrl+backspace=> delete the previous word
ctrl+r  => display cmd histroy
and so on

**Comment by kingToolbox** [comment]
Sorry for late reply.

Thanks for your advice. I have integrated `cmd` with `clink` and will release it in the next `WindTerm_2.4.0_Prerelease_2` version. Please stay tuned. :smile:

**Comment by JiangBoy** [comment]
> Sorry for late reply.
> 
> Thanks for your advice. I have integrated `cmd` with `clink` and will release it in the next `WindTerm_2.4.0_Prerelease_2` version. Please stay tuned. 😄

Thank you for creating such good-working software

**Comment by kingToolbox** [comment]
I am happy to inform you that [WindTerm_2.4.0_Prerelease_2](https://github.com/kingToolbox/WindTerm/releases/tag/2.4-prerelease) has been released and `clink` has been packaged. You need to execute menu item `Session pane - Right-click menu - Import - Shell Session` to regenerate the `cmd` and `admin:cmd` sessions. Please download and check it, thank you.

BTW:
You can also try the new feature [Free Type Mode](https://kingtoolbox.github.io/2022/04/12/free_type_mode/), which will be more convenient. I hope you like it.

**Comment by jan-bar** [comment]
I like cmder, and cmder also uses clink, but it provides richer functions

![image](https://user-images.githubusercontent.com/14202095/163111387-f04bcbc3-f1fc-441f-afae-916e31560cd9.png)

**Comment by kingToolbox** [comment]
I will solve the configuration problems of `folding` and `outlining` for you. But can the `init.bat` file of `Cmder` be used directly?

**Comment by jan-bar** [comment]
> I will solve the configuration problems of `folding` and `outlining` for you. But can the `init.bat` file of `Cmder` be used directly?

Yes, I use it in both jetbrains ide and vscode as shown below, The functions of cmder are more than these, because click is also used, I can also write lua scripts to implement custom command prompt functions

![image](https://user-images.githubusercontent.com/14202095/163292740-3c633e2b-32a9-4bb2-b5d2-0660c56903ef.png)

![image](https://user-images.githubusercontent.com/14202095/163292719-93947811-3233-4ae8-863c-b7e4dbd2e1bd.png)

**Comment by jan-bar** [comment]
> I will solve the configuration problems of `folding` and `outlining` for you. But can the `init.bat` file of `Cmder` be used directly?

Yes, I use it in both jetbrains ide and vscode as shown below, The functions of cmder are more than these, because click is also used, I can also write lua scripts to implement custom command prompt functions

![image](https://user-images.githubusercontent.com/14202095/163292740-3c633e2b-32a9-4bb2-b5d2-0660c56903ef.png)

![image](https://user-images.githubusercontent.com/14202095/163292719-93947811-3233-4ae8-863c-b7e4dbd2e1bd.png)

**Comment by kingToolbox** [comment]
I am happy to inform you that [WindTerm_2.4.0](https://github.com/kingToolbox/WindTerm/releases/tag/2.4.0) has been released and `folding` and `outling` of `Cmder` have been fixed. Please download and check it, thank you. 

BTW:
Currently `folding` and `outline` support the official prompt `λ` of `Cmder`, but may still not match your `(py39) #`. You can refer to the existing fragments to write your own matching rules.

Lexer: `WindTerm_Profiles_Path\terminal\schemes\cmd\cmd.lexer`
Folding: `WindTerm_Profiles_Path\terminal\schemes\cmd\cmd.fold`
Outlining: `WindTerm_Profiles_Path\terminal\schemes\cmd\cmd.outline`

If you have any questions, don't hesitate to ask me.

**Comment by jan-bar** [comment]
> 

Thanks a lot, it works fine. the `(py39) #` It's python's conda tool result , `#` is the custom I'm used to.

**Comment by kingToolbox** [comment]
Glad to hear this good news. There are indeed too many types of command prompts, and it is unrealistic to support all of them. The best way can only be customized by users.

The focus of `WindTerm` is still `SSH`, and because of cross-platform, specific development for a certain `Shell` is not yet possible. However, I think that with the development of plug-in features in the future, it should be as convenient as `Cmder` one day.

**Comment by jan-bar** [comment]
> Glad to hear this good news. There are indeed too many types of command prompts, and it is unrealistic to support all of them. The best way can only be customized by users.
> 
> The focus of `WindTerm` is still `SSH`, and because of cross-platform, specific development for a certain `Shell` is not yet possible. However, I think that with the development of plug-in features in the future, it should be as convenient as `Cmder` one day.

I have a good idea, to write a cross-platform executable program that interacts with the executable program when windterm connects to the server, so that I can easily get some required data. For example, `working directory`, `command history`, so that you don't have to match terminal text, and running efficiency should also improve. It is even possible to define only the specification, and the executable program is written by the user, which will be more flexible

Because I see an issue that wants the file manager to follow the current working directory of ssh, as well as the current command history. I don't know if this proposal is feasible.
